### ISSUE TITLE
Preprocessor climate_statistics fails

### ISSUE THREAD

**Issue by schlunma** [ESMValGroup/ESMValCore#694]
**Describe the bug**
After the merge of #684 the `climate_statistics` preprocessor does not work anymore. 

**Error message**
```
Traceback (most recent call last):
  File "/miniconda3/envs/mlr/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/ESMValCore/esmvalcore/_task.py", line 723, in _run_task
    output_files = task.run()
  File "/ESMValCore/esmvalcore/_task.py", line 242, in run
    self.output_files = self._run(input_files)
  File "//ESMValCore/esmvalcore/preprocessor/__init__.py", line 428, in _run
    product.apply(step, self.debug)
  File "/ESMValCore/esmvalcore/preprocessor/__init__.py", line 296, in apply
    self.cubes = preprocess(self.cubes, step, **self.settings[step])
  File "/ESMValCore/esmvalcore/preprocessor/__init__.py", line 241, in preprocess
    result.append(_run_preproc_function(function, item, settings))
  File "/ESMValCore/esmvalcore/preprocessor/__init__.py", line 224, in _run_preproc_function
    return function(items, **kwargs)
  File "/ESMValCore/esmvalcore/preprocessor/_time.py", line 399, in climate_statistics
    time_weights = get_time_weights(cube)
  File "//ESMValCore/esmvalcore/preprocessor/_time.py", line 176, in get_time_weights
    time_weights = time_thickness * da.ones_like(cube.data)
  File "/miniconda3/envs/mlr/lib/python3.8/site-packages/dask/array/core.py", line 1221, in __array_ufunc__
    return elemwise(numpy_ufunc, *inputs, **kwargs)
  File "/miniconda3/envs/mlr/lib/python3.8/site-packages/dask/array/core.py", line 3912, in elemwise
    broadcast_shapes(*shapes)
  File "/miniconda3/envs/mlr/lib/python3.8/site-packages/dask/array/core.py", line 3871, in broadcast_shapes
    raise ValueError(
ValueError: operands could not be broadcast together with shapes (120,) (120, 68, 180)
```

**Cause**

This is the problematic line:

https://github.com/ESMValGroup/ESMValCore/blob/df98aa0cf38492fe1eab63203bf3fa5447086191/esmvalcore/preprocessor/_time.py#L176

My cube has the shape `(120, 68, 180)` (=`time,lat,lon`) which can not be broadcasted to the shape of the time weights (`(120,)`).

Our tests do not fail because of pure chance. Either the tested cube have the "wrong" order (`lat,lon,time`) (with `time` as trailing dimension the broadcasting works) or the dimensions match because of luck.

**Fix**

I will create a fix using [`iris.util.broadcast_to_shape`](https://scitools.org.uk/iris/docs/latest/iris/iris).

**Comment by bouweandela** [comment]
>  Either the tested cube have the "wrong" order (lat,lon,time) 

It might be good to fix that test too, CMIP data does not look like that.

**Comment by schlunma** [comment]
Fixed tests in #695
